File: sdks/csharp/src/NovaBilling/Types/RegisterResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record RegisterResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("accessToken")]
    public required string AccessToken { get; set; }

    [JsonPropertyName("refreshToken")]
    public required string RefreshToken { get; set; }

    [JsonPropertyName("tenant")]
    public required TenantInfoResponse Tenant { get; set; }

    [JsonPropertyName("apiKey")]
    public required string ApiKey { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/register_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class RegisterResponse < Internal::Types::Model
      field :access_token, -> { String }, optional: false, nullable: false, api_name: "accessToken"
      field :refresh_token, -> { String }, optional: false, nullable: false, api_name: "refreshToken"
      field :tenant, -> { Novabilling::Types::TenantInfoResponse }, optional: false, nullable: false
      field :api_key, -> { String }, optional: false, nullable: false, api_name: "apiKey"
    end
  end
end
